app/api/help_seekers/needs_api.rb

```diff
@@ -120,6 +123,7 @@ class NeedsApi < Grape::API
 
             need.reviews.create!(review_params)
             need.closed!
+            need.update!(status_updated_at: DateTime.now, updated_by: current_user.id)
```

Comment:
Please take a look at the logs, my assumption is that these 2 lines will trigger 2 separate updates, we might consider having just one single update that includes the status update too.